### ISSUE TITLE
getContourPoint: Customizing to not display nodes on circumference

### ISSUE THREAD

**Issue by lokeshpkumar** [Muntaner/visjs_districts#2]
Hi, 

Great piece of work with this grouping of nodes, helping me a lot thank you!

I have one question though, in [this](https://github.com/visjs/vis-network/issues/203#issuecomment-584596684) link you mentioned that to change the circular disposition of nodes in a group we can customize the getContourPoint function. I needed to understand what formulae did you use to get the current x and y points and also could you please provide any other method using which we can derive other shapes/random points within the circular region?

**Comment by Muntaner** [comment]
Hello @lokeshpkumar ,
yes, as I mentioned in the comment the circular disposition of the nodes is determined by the [getContourPoint function](https://github.com/Muntaner/visjs_districts/blob/60e1975740c27a26b0b25dcc3c9f1bd259ca3405/js/districts.js#L15) in the example codebin I provided (_districts.js_ file).

As you can see, the function receives the following parameters:
- the **center** of the node's group, pre-calculated in the [partitionateRegion function](https://github.com/Muntaner/visjs_districts/blob/60e1975740c27a26b0b25dcc3c9f1bd259ca3405/js/districts.js#L1). For each group of the network you'll have a distinct (x,y) point, calculated in the function partitionateRegion by dividing the canvas in n regions where n is the number of distinct groups defined for the network. 
- the **range** of the region, tied to the size of the group. For a small group of region a small portion of the canvas will be used for the node's district, and instead a big one for a big group of nodes. As you can see, the range value is tampered in order not to become too big or too small (those values can be tuned in based on the size of the groups of your network). In the example the range is used for the circles of the districts, but I see no problems in using it for shaping other figures such as squares or polygons (big range -> big shape and viceversa). In this case the range corresponds to **the number of nodes per group**.
- the **angleIndex** . This index is different and incremental for each node of the network and is used to determine at which angle it should be placed. If, for example, we have 6 distinct nodes, we should divide a circle of 360° in 6 parts and place the nodes at the corresponding angles (0°, 60°, 120°, 180°, 240°, 300°). A single slice is 60°: node indexed 0 of the group will get angle 0° since index * slice = 0, and the same goes for the other nodes:

1. 60° * 1 = 60°
2. 60° * 2 = 120°
...
5. 60° * 5 = 300°

That's exactly what is calculated [in this part of the code](https://github.com/Muntaner/visjs_districts/blob/60e1975740c27a26b0b25dcc3c9f1bd259ca3405/js/districts.js#L16). What happens is a simple conversion from an angle to two coordinates by using [some trigonometry](https://github.com/Muntaner/visjs_districts/blob/60e1975740c27a26b0b25dcc3c9f1bd259ca3405/js/districts.js#L20).
- other parameters (**MIN_RANGE = 10, MAX_RANGE = 100, RANGE_MULTIPLIER = 10**) are used in the circle representation just to tamper the visualization and are not mandatory for the correct disposition of the nodes.

So, in short: you can rewrite and replace this function putting your own (x, y) positions per node. For example, for a square representation of the groups, I would imagine a row/columns matrix disposition of the nodes, calculating how many rows and columns are needed per region. I'll try to provide a working example of a box/rectangle districts representation when I'll have some free time.
